### PR TITLE
Re-adds the LWAP, reworked

### PR DESCRIPTION
**Pull request by Coolrune206**
## What Does This PR Do

Re-adds the LWAP sniper rifle to the game, with a rework to its functionality to better suit the current state of the game.
The LWAP sniper rifle functions similarly to the accelerator sniper rifle, starting with higher damage but increasing more slowly. Its benefits lie in its scope and ability to knock people down if the damage is above a certain threshold (40), with the knockdown scaling with the damage dealt.  It is limited by a severely bad firerate and gives you poor situational awareness while scoped. 
The LWAP used to appear in the Gamma Armory, but things that got the Gamma Armory caused were rarely something that would require the LWAP - so it is instead integrated into the RnD weapons kit system, requiring a LWAP kit and an Accelerator Laser (because it functions as a up/sidegrade to the Accel Laser). The LWAP kit requires Combat tech seven, Magnets tech seven, Powerstorage tech five, making it something that Science will actually have to work to produce.

For damage specifics, the LWAP starts at 10 damage and increments 5 damage per tile. Comparatively, the Accelerator Laser starts at 6 damage and increments 7 per tile, so it gets stronger considerably more quickly. The LWAP's strength is therefore in its ability to fire from further away with greater accuracy due to the scope.
## Why It's Good For The Game

The LWAP was removed in https://github.com/ParadiseSS13/Paradise/pull/17075, for good reason (as it was either terrible or overpowered depending on whether you were fighting a human mob or not). However, we could use more energy guns and the LWAP does have its own niche that it can fill. Converting it to knockdown and giving it range-based damage to complement its sights should hopefully help it find a good place in the current combat system.
## Images of changes

LWAP kit, unupgraded Protolathe
image
Ditto, upgraded Proto
image
Assembly
image
LWAP
image
LWAP kit
image
## Testing

Above images plus shot it a bunch, scratched head as to why knockdown wasn't working, realized dummies can't get knocked down, shot Vox (as god intended), knocked down when far enough away, life is good.
## Changelog
:cl:
add: Re-added the LWAP sniper rifle, with a rework
/:cl:

What the fuck does LWAP even stand for anyways?

### CODE REVIEW

**Comment by Samman166** [comment]
I'd suggest making it require some illegals levels maybe 3 just so it takes more than RnD doing what they do normally to actually be able to get it. Does it cap it's damage like accelerators cap out at 100? Also it's been a while so I gotta wonder what it's projectile speed is like since I kinda remember it being pretty slow, making the projectile just a bit faster might allow people to actually hit shots on moving targets.

**Comment by matttheficus** [comment]
How many shots does it have?
Can it fit in the Suit Storage slot or do you need to backpack it like the Ion Rifle?

**Comment by Coolrune206** [comment]
> How many shots does it have? 

Ten shots, five second delay between each.
> Can it fit in the Suit Storage slot or do you need to backpack it like the Ion Rifle?

It currently fits in Suit Storage, but that can be changed. Hopefully once it's in I'll be able to badger a spriter to give it a back/suit storage sprite.


> I'd suggest making it require some illegals levels maybe 3 just so it takes more than RnD doing what they do normally to actually be able to get it. 

I can if necessary, though I'd prefer if this was an option that could be acquired on most rounds that just requires help from botany (magnets 7) and usually the Captain and Robo (or botany) (combat 7) to make. Feel like if I gated this behind illegals it would have to be a bit more powerful.
> Does it cap it's damage like accelerators cap out at 100? 

Yup.
> Also it's been a while so I gotta wonder what it's projectile speed is like since I kinda remember it being pretty slow, making the projectile just a bit faster might allow people to actually hit shots on moving targets.

Projectile speed is unaffected from normal lasers, so it's not exactly slow. Speeding it up more might make it easier to hit moving targets, but considering you can already fire it at someone from off their screen using the scope, I think giving them a chance to dodge if they can is fair.
> Regardless I am happy to see LWAPs back and I like the niche you've got for them. Excited for their return

:)

File: code/modules/projectiles/guns/energy/laser.dm [hunk 1]
```diff
@@ -106,6 +106,46 @@
...
+	damage = 10
+
+/obj/item/projectile/beam/laser/sniper/Range()
+	..()
+	damage = min(damage+5, 100)
+
+/obj/item/projectile/beam/laser/sniper/on_hit(atom/target, blocked = 0, hit_zone)
+	..()
+	var/mob/living/carbon/human/M = target
+	if(ishuman(target) && damage >= 40)
```
Comment (lewcc) [inline code/modules/projectiles/guns/energy/laser.dm:146]:
```suggestion
	if(istype(M) && damage >= 40)
```

**Comment by SpringSkipper** [comment]
I'm kinda curious why this is in the game at all. Getting 1-shot softcritted and knocked down from offscreen doesn't sound particularly... fun? As far as a weapon for security to escalate towards, it seems ill-suited for powergaming vampires or maint-dwelling murderbone antags. I feel like this is just going to get used to 'style' on roleplay antags who make the mistake of standing still to type while no enemies are on screen.

**Comment by Coolrune206** [comment]
> I'm kinda curious why this is in the game at all. Getting 1-shot softcritted and knocked down from offscreen doesn't sound particularly... fun? As far as a weapon for security to escalate towards, it seems ill-suited for powergaming vampires or maint-dwelling murderbone antags. I feel like this is just going to get used to 'style' on roleplay antags who make the mistake of standing still to type while no enemies are on screen.

To be honest I'm mostly just re-adding this because it's really cool. 
I also think it will be a bit more effective in maint than you think, several hallways have significant length and it's harder to dodge in maints.
Lastly, I think the issue of sec using them on RP antags should be handled by sec's rules of lethals engagement. If the antag is generally RPing, sec probably doesn't have any reason (within space law) to just lethal them on-sight, and if sec DOES have a raeson to just shoot them, then they'd be able to... shoot them with anything.

**Review by S34NW — APPROVED** [review]

File: code/modules/projectiles/guns/energy/laser.dm [hunk 2]
```diff
@@ -106,6 +106,46 @@
...
+	zoomable = TRUE
+	zoom_amt = 7
+	shaded_charge = TRUE
+
+/obj/item/ammo_casing/energy/laser/sniper
+	projectile_type = /obj/item/projectile/beam/laser/sniper
+	muzzle_flash_color = LIGHT_COLOR_PINK
+	select_name = "sniper"
+	fire_sound = 'sound/weapons/marauder.ogg'
+	delay = 50
```
Comment (hal9000PR) [inline code/modules/projectiles/guns/energy/laser.dm:131]: use seconds

**Review by SteelSlayer — APPROVED** [review]

**Comment by Coolrune206** [comment]
lwap

**Pull request merged**